### ISSUE TITLE
Faraday's `open_timeout` not configurable and defaults to 60 seconds!

### ISSUE THREAD

**Issue by andrewferk** [Azure/azure-ruby-asm-core#30]
If for any reason the Azure Storage service cannot be reached, the Faraday client will not timeout until after 60 seconds and this is not configurable.

`open_timeout` is set in [azure-ruby-asm-core](https://github.com/Azure/azure-ruby-asm-core/blob/master/lib/azure/http_response_helper.rb#L34), and I don't know of I way to have it already set. I would prefer this to be defined by an environment variable.

It looks like it was [addressed in April](https://github.com/Azure/azure-sdk-for-ruby/pull/335), but for some reason it was decided 60 seconds was a good timeout value.

**Comment by sarangan12** [comment]
@yaxia Can you take a look at this issue and share your thoughts about making this value configurable?

**Comment by yaxia** [comment]
The value is configurable from the `option` parameter when creates the Faraday object. In my opinion, either read the value from an option or an environment variable is OK. But this should be the call of the implementation of the HTTP [client](https://github.com/Azure/azure-ruby-asm-core/blob/master/lib/azure/core/service.rb#L34). 

Since the `open_timeout` is more like a per connection setting instead of a global setting from a library's perspective, I'd prefer keep the current implementation and leave the flexibility to the user to decide how the client will provide the connection options, setting from code or reading from environment variables. As a reference, the Azure Storage [client](https://github.com/Azure/azure-storage-ruby/blob/master/lib/azure/storage/core/http_client.rb#L57) implements its own way of creating a Faraday object, by reading the CA file path from code and proxy settings from the environment variables.

BTW, the default value of 60 seconds is from the Ruby implementation, [NET::Http, open_tiumeout attribute](http://ruby-doc.org/stdlib-2.4.0/libdoc/net/http/rdoc/Net/HTTP.html).

+@veronicagg for more comment.

**Comment by veronicagg** [comment]
I agree that we're currently not providing a way to configure `open_timeout`. We're not planning to address this issue soon though, we'd be happy to review a PR if one is submitted.

**Comment by OpakAlex** [comment]
`not providing a way to configure open_timeout`
Так на хуй вы писали эту пизду?
